test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -549,3 +609,60 @@ func runReplicaSet(config testutils.ReplicaSetConfig) error {
 	config.ContainerDumpFunc = e2ekubectl.LogFailedContainers
 	return testutils.RunReplicaSet(config)
 }
+
+// CreateContainerResourceCPUHorizontalPodAutoscaler create a horizontal pod autoscaler with container resource target
+// for consuming resources.
+func CreateContainerResourceCPUHorizontalPodAutoscaler(rc *ResourceConsumer, cpu, minReplicas, maxRepl int32) *autoscalingv2beta2.HorizontalPodAutoscaler {
+	hpa := &autoscalingv2beta2.HorizontalPodAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rc.name,
+			Namespace: rc.nsName,
+		},
+		Spec: autoscalingv2beta2.HorizontalPodAutoscalerSpec{
+			ScaleTargetRef: autoscalingv2beta2.CrossVersionObjectReference{
+				APIVersion: rc.kind.GroupVersion().String(),
+				Kind:       rc.kind.Kind,
+				Name:       rc.name,
+			},
+			MinReplicas: &minReplicas,
+			MaxReplicas: maxRepl,
+			Metrics: []autoscalingv2beta2.MetricSpec{
+				{
+					Type: "ContainerResource",
+					ContainerResource: &autoscalingv2beta2.ContainerResourceMetricSource{
+						Name:      "cpu",
+						Container: rc.name,
+						Target: autoscalingv2beta2.MetricTarget{
+							Type:               "Utilization",
+							AverageUtilization: &cpu,
+						},
+					},
+				},
+			},
+		},
+	}
+	hpa, errHPA := rc.clientSet.AutoscalingV2beta2().HorizontalPodAutoscalers(rc.nsName).Create(context.TODO(), hpa, metav1.CreateOptions{})
+	framework.ExpectNoError(errHPA)
+	return hpa
+}
+
+// DeleteContainerResourceHPA delete the horizontalPodAutoscaler for consuming resources.
+func DeleteContainerResourceHPA(rc *ResourceConsumer, autoscalerName string) {
+	rc.clientSet.AutoscalingV2beta2().HorizontalPodAutoscalers(rc.nsName).Delete(context.TODO(), autoscalerName, metav1.DeleteOptions{})
+}
+
+//SidecarStatusType type for sidecar status
+type SidecarStatusType bool
+
+var (
+	Enable  SidecarStatusType = true
+	Disable SidecarStatusType = false
+)
```

Comment:
I would expect these variables to be constants.

Reply:
Changed them to constants